### PR TITLE
#166107267 Enable Users Edit and Delete their articles

### PR DESCRIPTION
**Pull request by patrickf949**
### What does this PR do?

Enables users edit and delete only artilces they created.

#### Description of Task to be completed?

- Add page for users to edit articles they created
- Add functionality of deleting an article

#### How should this be manually tested?
Clone the repo

- `$ git clone https://github.com/andela/ah-frontend-prime.git`

Check into the project directory and Check out the branch  `ft-edit-delete-articles-166107267`

- `$ cd ah-frontend-prime`
- `$ git checkout  ft-edit-delete-articles-166107267`

Install all dependencies

- `$ npm install`

Start the server

- `$npm start`

Create an acount and login with the sign up and sign in buttons on the dashboard

Create an article By hovering on the picture in the right corner and clicking create article.
Add all necessary fields

View the articles you created by hovering over the profile picture and clicking `Your stories`

Click edit on whichever article you'd like to edit to be provided a form in which you'll enter all the necessary changes you'd like to make to edit the article.

Click delete to delete an article amongst your stories.

#### What are the relevant pivotal tracker stories?
[#166107267](https://www.pivotaltracker.com/story/show/166107267)

#### Screenshots (if appropriate)
Create Article

Screenshot 2019-05-26 at 16 43 12

User Articles

Screenshot 2019-05-26 at 16 43 24

Edit Article Page

Screenshot 2019-05-26 at 16 43 44

Delete Article

Screenshot 2019-05-26 at 16 44 35

### CODE REVIEW

**Review by anyatibrian — COMMENTED** [review]
LGTM

File: src/components/articles/editArticlePage.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  componentWillReceiveProps(newProps) {
+    if (newProps) {
+      this.setState({
+        title: newProps.article.article.title,
+        description: newProps.article.article.description,
+        body: newProps.article.article.body,
+        image: newProps.article.article.image,
+        tags: newProps.article.tagList,
+        article: newProps.article.article
+      });
```
Comment (Genza999) [inline src/components/articles/editArticlePage.js:39]: Hey @patrickf949 , kindly destructure the newProps here.

**Review by e-ian — APPROVED** [review]
@patrickf949 just work on the destructuring as @Genza999. Otherwise great work. LGTM

**Review by MuhanguziDavid — APPROVED** [review]
Nice work @patrickf949 . LGTM

**Pull request merged**